pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
@ghodss incompatible change constraints, yes that kind of thing could just be implemented in the server-side validation of the final state of the objects post-patch. It properly belongs there, and should not be reimplemented here, to your point.